core/tests/recovery-test/tests/snapshot-recovery.test.ts

```diff
@@ -160,17 +162,23 @@ describe('snapshot recovery', () => {
                 }
                 sampledCount++;
 
-                const snapshotAccountAddress = '0x' + storageLog.accountAddress.toString('hex');
-                const snapshotKey = '0x' + storageLog.storageKey.toString('hex');
-                const snapshotValue = '0x' + storageLog.storageValue.toString('hex');
                 const snapshotL1BatchNumber = storageLog.l1BatchNumberOfInitialWrite;
-                const valueOnBlockchain = await mainNode.getStorageAt(
-                    snapshotAccountAddress,
-                    snapshotKey,
-                    l2BlockNumber
-                );
-                expect(snapshotValue).to.equal(valueOnBlockchain);
                 expect(snapshotL1BatchNumber).to.be.lessThanOrEqual(l1BatchNumber);
+
+                if (snapshotMetadata.version === 0) {
+                    const snapshotAccountAddress = '0x' + storageLog.accountAddress.toString('hex');
+                    const snapshotKey = '0x' + storageLog.storageKey.toString('hex');
+                    const snapshotValue = '0x' + storageLog.storageValue.toString('hex');
+                    const valueOnBlockchain = await mainNode.getStorageAt(
+                        snapshotAccountAddress,
+                        snapshotKey,
+                        l2BlockNumber
+                    );
+                    expect(snapshotValue).to.equal(valueOnBlockchain);
+                } else {
+                    // TODO: requires adding a method like `zks_resolveStorageAt(hashedKey)` returning
+                    //   { address, key, enumerationIndex }
```

Comment:
Would such a method be useful?

Reply:
I think it's better not to add any new methods unless there is a strong business or technical need for it.
Right now I don't think this method will bring much value.